modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCartButton.js

```diff
@@ -58,60 +125,25 @@ function AddToCartButton({
 			disabled={disabled}
 			displayType="primary"
 			monospaced={settings.iconOnly && settings.inline}
-			onClick={(event) => {
-				if (cartAtomState.updating) {
-					return;
-				}
+			onClick={async (event) => {
+				const CartResource = ServiceProvider.DeliveryCartAPI('v1');
+				const order = await CartResource.getCartsByAccountIdAndChannelId(
+					accountId,
+					channel.id
+				);
+
+				if (showOrderTypeModal && !order.items.length) {
+					setEvent(event);
+					Liferay.fire(OPEN_MODAL, {
```

Comment:
Are we sure it doesn't require a prefix such as a portlet namespace? What if we have more instances of the same portlet?

Reply:
We can keep it like this because it won't open another modal, just the one to select order type

Reply:
Can we test the behavior with 2 portlets on the same page, I have the feeling that 2 modals will be opening

Reply:
Hey @alessiorendina 
I tested it and it does not open the 2 at the same time.